### PR TITLE
CSI: Add LogLevel for csi templates

### LINKED ISSUE #4690: Make CSI verbosity configurable
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
We should be able to specify a log level to set for CSI components
**What is use case behind this feature:**
Currently CSI components produce a lot of logs (mostly `csi-snapshotter` and `csi-rbdplugin` in my case), it would be nice to have a way to manage it.

### PR DESCRIPTION
**Pull request by Madhu-1**
**Description of your changes:**
Currently, there is no option to turn on and off the log level verbosity in csi containers, This PR adds the functionality to select logging levels for csi containers.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

**Which issue is resolved by this Pull Request:**
Resolves #4690

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.
[test ceph]

### CODE REVIEW

File: pkg/operator/ceph/csi/spec.go [hunk 1]
```diff
@@ -243,6 +248,17 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 		deployProvSTS = true
 	}
 
+	logLevel := os.Getenv("CSI_LOG_LEVEL")
+	tp.LogLevel = defaultLogLevel
+	if logLevel != "" {
+		l, err := strconv.ParseUint(logLevel, 10, 8)
+		if err != nil {
+			logger.Errorf("Failed to parse CSI_LOG_LEVEL err: %+v .Defaulting to %d", err, defaultLogLevel)
```
Comment (leseb) [inline pkg/operator/ceph/csi/spec.go]:
```suggestion
			logger.Errorf("Failed to parse CSI_LOG_LEVEL. Defaulting to %d. %v", defaultLogLevel, err)
```

File: pkg/operator/ceph/csi/spec.go [hunk 2]
```diff
@@ -127,6 +129,9 @@ const (
 	DefaultCephFSLivenessMerticsPort uint16 = 9081
 	DefaultRBDGRPCMerticsPort        uint16 = 9090
 	DefaultRBDLivenessMerticsPort    uint16 = 9080
+
+	// default log level for csi containers
+	defaultLogLevel uint8 = 5
```
Comment (leseb) [inline pkg/operator/ceph/csi/spec.go]: Shouldn't this be 0? It looks quite aggressive to use debug logs.

**Comment by Madhu-1** [comment]
@leseb updated PR, PTAL

File: pkg/operator/ceph/csi/spec.go [hunk 3]
```diff
@@ -243,6 +248,17 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 		deployProvSTS = true
 	}
 
+	logLevel := os.Getenv("CSI_LOG_LEVEL")
+	tp.LogLevel = defaultLogLevel
+	if logLevel != "" {
+		l, err := strconv.ParseUint(logLevel, 10, 8)
+		if err != nil {
+			logger.Errorf("Failed to parse CSI_LOG_LEVEL. Defaulting to %d. %v", defaultLogLevel, err)
```
Comment (leseb) [inline pkg/operator/ceph/csi/spec.go]:
```suggestion
			logger.Errorf("failed to parse CSI_LOG_LEVEL. defaulting to %d. %v", defaultLogLevel, err)
```

**Comment by Madhu-1** [comment]
@leseb done PTAL

**Comment by Madhu-1** [comment]
can we review this one after https://github.com/rook/rook/pull/4726 ?

**Comment by Madhu-1** [comment]
@leseb @travisn PTAL

**Comment by travisn** [comment]
> Do we have a doc section where we could put this for CSI? Maybe troubleshooting?

@Madhu-1 How about adding a new section in ceph-common-issues.md for how to troubleshoot the CSI driver? It would be good to include more than just how to increase the log level. I'll go ahead and merge this and let's follow up with a new PR, thanks.

**Pull request merged**